standards/peer-review.md

```diff
@@ -1,8 +1,33 @@
+## Pull Request Workflow
+
+- All Pull Requests to NYPL repositories MUST be reviewed prior to merging (caveat below).
+  - PRs MUST NOT be merged if a reviewer requests changes
+  - Teams MAY agree to specific written policies as to conditions when reviews are not required. Such a policy MUST be approved by an Engineering Manager or Director.
+  
+- Engineers MUST mark at least one (1) developer as a reviewer and SHOULD request reviews from all relevant engineers
+  - Reviewers MAY come from any team within NYPL Digital
+  - Teams MAY create [CODEOWNERS](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) files to establish a default group of reviewers
+  - Experience SHOULD NOT be taken into account when requesting review
+  - Requests for specific feedback MAY be requested by tagging specific engineers in PR comments
+
+- Engineers marked as reviewers SHOULD complete the review in a timely manner
+  - This period SHOULD be agreed upon within individual development teams, 24-48 hours has proven to be a reasonable starting place
+  - Code review SHOULD take some time (more than a quick LGTM) but should not take more than 1-2 hours
+  - If a code review is very lengthy or complex, reviewers MAY request that the PR be broken up into smaller pieces and/or request a walkthrough/live review session
+ 
+- If changes are requested (or made in response to questions) the requesting engineer SHOULD re-request review when ready
```

Comment:
Something that we do in the DS repo is to add labels to PRs that need a review "Needs Review", need changes "Changes Requested", or are approved "Ship It". This helps us glance at the PR page and see what PRs we should review. I think we can suggest to add appropriate labels to PRs if they have been discussed within the team.

For example:
<img width="327" alt="Screenshot 2024-04-17 at 4 21 42 PM" src="https://github.com/NYPL/engineering-general/assets/1280564/a4ed5606-f877-489c-a86c-f3d1a6032134">

Reply:
This would be a responsibility of the requestor to add the initial label, and the reviewer to update the label when they complete their review.

Reply:
I've added this as an option! I think this is a great idea for larger, more asynchronous teams (such as DS and eReading) but for some smaller teams may not be necessary. We should keep an eye on this and see if we want to make a firmer recommendation in the future as this specific issue was one of my motivations for updating this page.